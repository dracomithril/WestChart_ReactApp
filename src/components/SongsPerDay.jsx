/**
 * Created by Gryzli on 29.03.2017.
 */
import React from "react";
import PropTypes from 'prop-types';
import {Panel, Popover, Tooltip, OverlayTrigger} from "react-bootstrap";
export default class SongsPerDay extends React.Component {
    render() {
        const {store} = this.context;
        const {songs_per_day} = store.getState();
        const errorDays = this.props.error_days;
        let DayEntry = (props) => <div key={props.org}>
        <span style={{color: props.color}}
        >{new Date(props.org).toDateString()}</span>
        </div>;

        const err_days_less = errorDays.filter((elem) => elem.color === 'blue').map((elem) => DayEntry(elem));
        const err_days_more = errorDays.filter((elem) => elem.color === 'red').map((elem) => DayEntry(elem));

        let header = (<div>
            <strong style={{paddingRight: 20}}>{'songs per day'}</strong>
            <input type="number" value={songs_per_day} className="num_days"
                   name="songs_per_day" step={1} max={10} min={1}
                   onChange={(e) => store.dispatch({
                       type: 'UPDATE_SONGS_PER_DAY',
                       days: Number(e.target.value)
                   })}/>
        </div>);
        let less_header = "less then expected";
        const popoverLess = (<Popover id="haveLessDays" title={less_header}>
            {err_days_less}
        </Popover>);
        const tooltipLess = <Tooltip id="haveLessDays">{less_header}</Tooltip>;
        let more_header = "more then expected";
        const popoverMore = (<Popover id="haveMoreDays" title={more_header}>
            {err_days_more}
        </Popover>);
        const tooltipMore = <Tooltip id="haveMoreDays">{more_header}</Tooltip>;
        let footer = (<div>
            {"errors count: " + errorDays.length}
            {err_days_less.length > 0 && <OverlayTrigger trigger={['hover', 'focus']} placement="bottom"
                                                         overlay={err_days_less.length > 0 ? popoverLess : tooltipLess}>
                <i style={{color: 'blue', marginRight: 5, marginLeft: 5}}
                   className="fa fa-arrow-circle-down" aria-hidden="true">{err_days_less.length}</i>
            </OverlayTrigger>}
            {err_days_more.length > 0 && <OverlayTrigger trigger={['hover', 'focus']} placement="bottom"
                                                         overlay={err_days_more.length > 0 ? popoverMore : tooltipMore}>
                <i style={{color: 'red', marginLeft: 5}} className="fa fa-arrow-circle-up"
                   aria-hidden="true">{err_days_more.length}</i></OverlayTrigger>}
        </div>);
        return (<Panel style={{float: 'left'}} header={header}
                       bsStyle={errorDays.length !== 0 ? "danger" : "success"}>
            {footer}
        </Panel>);
    }
}
SongsPerDay.contextTypes = {
    store: PropTypes.object
};
SongsPerDay.propTypes = {
    error_days: PropTypes.array
};