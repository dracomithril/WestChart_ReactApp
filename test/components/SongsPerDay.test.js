/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import SongsPerDay from './../../src/components/SongsPerDay';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('./../data/initial_state.json');

const mockStore = configureMockStore([]);
describe('<SongsPerDay/>', () => {
    it('renders without crashing', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <SongsPerDay error_days={[]}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });
    xit('renders without crashing', () => {
        const err_per_da= [{org:"2017/05/02"}];
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <SongsPerDay error_days={err_per_da}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        // let children =wrapper.find('div#error_list');
        // expect(children.children().key()).toBe("2017/05/02");

    });
});
