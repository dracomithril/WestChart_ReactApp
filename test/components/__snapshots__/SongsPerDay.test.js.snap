// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongsPerDay/> renders without crashing 1`] = `
<Panel
  bsClass="panel"
  bsStyle="success"
  defaultExpanded={false}
  footer={false}
  header={
    <div>
      <strong
        style={
          Object {
            "paddingRight": 20,
          }
        }
      >
        songs per day
      </strong>
      <input
        className="num_days"
        max={10}
        min={1}
        name="songs_per_day"
        onChange={[Function]}
        step={1}
        type="number"
      />
    </div>
  }
/>
`;

exports[`<SongsPerDay/> renders without crashing 2`] = `
<Panel
  bsClass="panel"
  bsStyle="danger"
  defaultExpanded={false}
  footer="miss mach count: 1"
  header={
    <div>
      <strong
        style={
          Object {
            "paddingRight": 20,
          }
        }
      >
        songs per day
      </strong>
      <input
        className="num_days"
        max={10}
        min={1}
        name="songs_per_day"
        onChange={[Function]}
        step={1}
        type="number"
      />
    </div>
  }
>
  <div>
    <strong>
      Date:
    </strong>
    (
    <span
      style={
        Object {
          "color": "blue",
          "marginRight": 5,
        }
      }
    >
      not enough
    </span>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      to many
    </span>
    )
    <br />
    <div
      id="error_list"
    >
      <div>
        <span
          style={
            Object {
              "color": undefined,
            }
          }
        >
          Tue May 02 2017
        </span>
      </div>
    </div>
  </div>
</Panel>
`;
